dashdotdb/models/deploymentvalidation.py

```diff
@@ -0,0 +1,52 @@
+from dashdotdb.models.base import db
+
+
+class ValidationToken(db.Model):
+    __tablename__ = 'validationtoken'
+    id = db.Column(db.Integer, primary_key=True)
+    timestamp = db.Column(db.DateTime)
+    deploymentvalidation = db.relationship('DeploymentValidation',
+                                           backref='validationtoken')
+
+
+class DeploymentValidation(db.Model):
+    __tablename__ = 'deploymentvalidation'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(256), unique=False)
+    token_id = db.Column(db.Integer, db.ForeignKey('validationtoken.id'))
+    namespace_id = db.Column(db.Integer, db.ForeignKey('dvnamespace.id'))
+    objectkind_id = db.Column(db.Integer, db.ForeignKey('objectkind.id'))
+    validation_id = db.Column(db.Integer, db.ForeignKey('validation.id'))
+
+
+class DVNamespace(db.Model):
+    __tablename__ = 'dvnamespace'
```

Comment:
We have a `namespace` table and a `cluster` table already. Why do we need new ones?
Anyway, if that turns out to be really required, the ERD needs update.

Reply:
Initially started out re-using this, but it got quite messy with conflicts from the overlapping db model classes. tried a few options to either extend an existing table as well as rebased to inherit a base classs.. and it was not very simple to manage or reference. ultimately this isn't a ton of data, so having distinct tables here is okay. the schema for the dvo metrics will likely change a bit as we iterate on it anyway. i've split off erd into deploymentvalidation and imagemanifestvuln to represent the schemas.

Reply:
Have you tried my initial suggestion of merging the two models in one? That really seems the right thing to do...

Reply:
yeah. i mean, i could do that, but it means re-writing places where the current imagemanifestvuln class(es) appear into tbd-newclass(es). this mr as it sits now works. i wanted to approach this from how a tenant would add to dddb. it would be good to eventually move this fully into app-interface like we did with app-sre-observability.

Reply:
I don't know where that use case came from... not aware about the requirements there, but I doubt that creating more - redundant - tables in the same database is the way to achieve multi-tenancy. If that's a use-case we want to cover now, we should probably get back to the whiteboard.

Reply:
Sorry, perhaps I am/was not in sync with what the DDDB piece full intent is. I had thought this service was intended to be more than just exclusive to app-sre in scope, following other self-service pieces. I had hoped to not have to re-write the existing class(es) and avoid each addition here requiring knowledge of the existing implementations (especially to ensure we don't end up with concurrency problems). In order to combine the models into one, I'll have to re-implement several of the classes and (possibly) some of the other pieces that depend on that model (maybe not, if the model can just be re-inherited under the existing class names). All the existing code is very specific to the ImageVuln model. Likely I'll have to use the single table inheritance to ensure the generated relational queries work properly. For instance, for a cascading delete of a an IVM we wouldn't want it to remove the cluster/namespace rows where those are also consumed by a DV. (https://docs.sqlalchemy.org/en/13/orm/inheritance.html#single-table-inheritance). 

For today, I'd like to get the current code merged in so I can at least get the general reporting for DVO end to end tested and tangible (at least functional). The DV data is very minimal in terms of data it will consume today. I'll timebox the monolithic model and if it seems like it going to take more than an hour or two, I can get the reporting for DV deployed and then immediately follow up with a re-write of the models and dependent classes.. and use some new name that isn't ImageVulnManifest or DeploymentValidation for the base class.